text/0000-supertrait-item-shadowing-v2.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: `supertrait_item_shadowing`
+- Start Date: 2024-05-04
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+When name resolution encounters an ambiguity between 2 trait methods, if one trait is a sub-trait of the other then select that method instead of reporting an ambiguity error.
```

Comment:
I know it's implicit in the statement of "encounters an ambiguity" (because having an ambiguity is dependent on actually having two methods in scope), but I would appreciate if the RFC mentions that we won't automatically select a subtrait method if it's not in scope.

In practice, for the `Itertools` example, this RFC's new rules only apply if we actually `use itertools::Itertools` so it's actually in scope (and therefore it's able to create an ambiguity at all).